code/game/machinery/computer/security.dm

```diff
@@ -34,12 +22,89 @@
 			O.forceMove(src)
 			scanner = O
 			to_chat(user, "You insert [O].")
+			update_static_data(usr)
 
 	. = ..()
 
+/obj/structure/machinery/computer/secure_data/ui_static_data(mob/user)
+	. = ..()
+
+	if(!allowed(usr))
+		to_chat(user, SPAN_WARNING("Access denied."))
+		return
+
+	// Pass player data, only for frontend state
+	var/mob/living/carbon/human/U = usr
+	var/user_data = list("name" = U.get_authentification_name(), "rank" = U.get_assignment())
+	.["user_data"] = user_data
+
+	// Pass scanner data
+	var/list/prints_data = list()
+	if (scanner && scanner.print_list)
+		for (var/obj/effect/decal/prints/prints in scanner.print_list)
+			var/print_entry = list(
+				"name" = prints.criminal_name,
+				"squad" = prints.criminal_squad,
+				"rank" = prints.criminal_rank,
+				"description" = prints.description
+			)
+			prints_data += list(print_entry)
+
+	var/scanner_status = list(
+		"connected" = !!scanner,
+		"count" = scanner && scanner.print_list ? length(scanner.print_list) : 0,
+		"data" = prints_data,
+	)
+	.["scanner"] = scanner_status
+
+	// Define criminal statuses and colors
+	var/list/criminal_statuses = list(
+	"*Arrest*" = list("background" = "#990c28", "font" = "#ffffff"),
+	"Incarcerated" = list("background" = "#faa20a", "font" = "#ffffff"),
+	"Released" = list("background" = "#2981b3", "font" = "#ffffff"),
+	"Suspect" = list("background" = "#686A6C", "font" = "#ffffff"),
+	"NJP" = list("background" = "#b60afa", "font" = "#ffffff"),
+	"None" = list("background" = "inherit", "font" = "inherit"),
+	)
+
+	.["criminal_statuses"] = criminal_statuses
+
+	// Map security records via id
+	var/list/records = list()
+	var/list/security_map = list()
+	for (var/datum/data/record/security in GLOB.data_core.security)
+		security_map[security.fields["id"]] = security
+
+	for (var/datum/data/record/general in GLOB.data_core.general)
```

Comment:
im pretty sure datacore gets added to during the round as people spawn in after round start